lib/run-context.js

```diff
@@ -119,7 +119,9 @@ RunContext.prototype._run = function () {
  */
 RunContext.prototype.toPromise = function () {
   return new Promise(function (resolve, reject) {
-    this.on('end', resolve);
+    this.on('end', function () {
+      resolve(this.targetDirectory);
```

Comment:
`this` is undefined in this case, you'll need to bind a context to this function.